### PR TITLE
Discord database integration

### LINKED ISSUE #45: Add Database integration with Discord ID
### Pre-requisites

- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find an existing bug.

### Describe the solution you'd like

The User table and User entity need to have a field for a Discord ID to allow us to link their account.
This field should be nullable.

The Discord login flow should add the returned user ID to the user row and redirect to the community page and indicate success in some way (change the button to indicate user has already linked their Discord).

### Describe alternatives you've considered

N/A

### PR DESCRIPTION
**Pull request by TevinBeckwith**
### Pre-Requisites

- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevant code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### Description of Changes

This PR includes:
- a migration file to add discordId to the user table
- changes to the discord.ts file that saves the retrieved discord id to the current user in the database.
- tests for the above changes
-
### Related Issues

Closes #45

### CODE REVIEW

File: packages/api/src/api/auth/discord.ts [hunk 1]
```diff
@@ -58,21 +49,19 @@ discord.get('/discord/callback', async (req, res) => {
 
       const { id: discordId } = await userResponse.json();
 
-      const { passport } = req.session as unknown as SessionData;
-      const { user: userGithubId } = passport as PassportData;
-      const user = await req.entityManager.findOne(User, { githubId: userGithubId });
+      const user = await req.entityManager.findOne(User, { githubId: req.user });
```
Comment (johnkahn) [inline packages/api/src/api/auth/discord.ts:52]: I believe it's going to be under `id` 🤔 

```suggestion
      const user = await req.entityManager.findOne(User, { githubId: req.user?.id });
```

**Review by johnkahn — CHANGES_REQUESTED** [review]
Quick change, if that's not the case, just resolve it. But I'm pretty sure `req.user` is an object with the fields inside of it

File: packages/api/src/api/auth/discord.ts (hunk 1 above)
Comment (TevinBeckwith) [reply packages/api/src/api/auth/discord.ts:52]: I tested it, and the req.user just returns a sting (githubId).  Maybe this is an issue that needs to be changed?

Comment (johnkahn) [reply packages/api/src/api/auth/discord.ts:52]: Maybe it's in another PR that hasn't been merged yet 🤔 We can merge this then

**Comment by johnkahn** [comment]
Just merged #50 which came with the changes to the User interface. It should be `req.user.profile.id` now

**Review by johnkahn — APPROVED** [review]
LGTM

**Pull request merged**